composer/trainer/trainer.py

```diff
@@ -2240,7 +2243,7 @@ def _train_microbatch(self, use_grad_scaling: bool, current_batch_size: int,
         device_batch = deepcopy(self.state.batch)
 
         microbatch_num_samples = self._train_data_spec.get_num_samples_in_batch(self.state.batch)
-        if self.deepspeed_enabled:
+        if self.deepspeed_enabled or not isinstance(self.state.model, DistributedDataParallel):
```

Comment:
were you intending to check if the model is FSDP here (this is what the pr comment says)?

Reply:
Sorry, it should be non-DDP. PR comment is bad (I assuemd non-DDP = FSDP) but it should be non-DDP since it includes no parallelism